packages/contracts-bedrock/tasks/check-l2.ts

```diff
@@ -0,0 +1,628 @@
+    if (owner === hre.ethers.constants.AddressZero) {
+      throw new Error('owner misconfigured')
+    }
+    logLoud()
+    console.log(`  - owner: ${owner}`)
+    logLoud()
+
+    const MESSAGE_VERSION = await L2CrossDomainMessenger.MESSAGE_VERSION()
+    console.log(`  - MESSAGE_VERSION: ${MESSAGE_VERSION}`)
+    const MIN_GAS_CALLDATA_OVERHEAD =
+      await L2CrossDomainMessenger.MIN_GAS_CALLDATA_OVERHEAD()
+    console.log(`  - MIN_GAS_CALLDATA_OVERHEAD: ${MIN_GAS_CALLDATA_OVERHEAD}`)
+    const MIN_GAS_CONSTANT_OVERHEAD =
+      await L2CrossDomainMessenger.MIN_GAS_CONSTANT_OVERHEAD()
+    console.log(`  - MIN_GAS_CONSTANT_OVERHEAD: ${MIN_GAS_CONSTANT_OVERHEAD}`)
+    const MIN_GAS_DYNAMIC_OVERHEAD_DENOMINATOR =
+      await L2CrossDomainMessenger.MIN_GAS_DYNAMIC_OVERHEAD_DENOMINATOR()
+    console.log(
+      `  - MIN_GAS_DYNAMIC_OVERHEAD_DENOMINATOR: ${MIN_GAS_DYNAMIC_OVERHEAD_DENOMINATOR}`
+    )
+    const MIN_GAS_DYNAMIC_OVERHEAD_NUMERATOR =
+      await L2CrossDomainMessenger.MIN_GAS_DYNAMIC_OVERHEAD_NUMERATOR()
+    console.log(
+      `  - MIN_GAS_DYNAMIC_OVERHEAD_NUMERATOR: ${MIN_GAS_DYNAMIC_OVERHEAD_NUMERATOR}`
+    )
+
+    const slot = await hre.ethers.provider.getStorageAt(
+      predeploys.L2CrossDomainMessenger,
+      0
+    )
+
+    const spacer = '0x' + slot.slice(26)
+    console.log(`  - legacy spacer: ${spacer}`)
+
+    const initialized = '0x' + slot.slice(24, 26)
+    if (initialized !== '0x01') {
+      throw new Error('not initialized')
+    }
+    console.log(`  - initialized: ${initialized}`)
+  },
+  // GasPriceOracle
+  // - check version
+  // - check decimals
+  GasPriceOracle: async (hre: HardhatRuntimeEnvironment) => {
+    const GasPriceOracle = await hre.ethers.getContractAt(
+      'GasPriceOracle',
+      predeploys.GasPriceOracle
+    )
+
+    const version = await GasPriceOracle.version()
+    assertSemver(version, 'GasPriceOracle')
+
+    const decimals = await GasPriceOracle.decimals()
+    if (!decimals.eq(6)) {
+      throw new Error('decimals misconfigured')
+    }
+    console.log(`  - decimals: ${decimals.toNumber()}`)
+
+    await checkProxy(hre, 'GasPriceOracle')
+    await assertProxy(hre, 'GasPriceOracle')
+  },
+  // L2StandardBridge
+  // - check version
+  L2StandardBridge: async (hre: HardhatRuntimeEnvironment) => {
+    const L2StandardBridge = await hre.ethers.getContractAt(
+      'L2StandardBridge',
+      predeploys.L2StandardBridge
+    )
+
+    const version = await L2StandardBridge.version()
+    assertSemver(version, 'L2StandardBridge', '0.0.2')
+
+    const OTHER_BRIDGE = await L2StandardBridge.OTHER_BRIDGE()
+    if (OTHER_BRIDGE === hre.ethers.constants.AddressZero) {
+      throw new Error('invalid OTHER_BRIDGE')
+    }
+    logLoud()
+    console.log(`  - OTHER_BRIDGE: ${OTHER_BRIDGE}`)
+    logLoud()
+
+    const MESSENGER = await L2StandardBridge.MESSENGER()
+    if (MESSENGER !== predeploys.L2CrossDomainMessenger) {
```

Comment:
You could use the `assertContractVariable` helper function in `src/deploy-utils` to simplify a few things

Reply:
This is a nice util but i need to log the values